src/Text.Analyzers/UnitTests/IdentifiersShouldBeSpelledCorrectlyTests.cs

```diff
@@ -1,8 +1,531 @@
+        {
+            var source = @"class Clazz { const string Naem = ""foo""; }";
+            var xmlDictionary = CreateXmlDictionary(new[] { "clazz" });
+            var dicDictionary = CreateDicDictionary(new[] { "naem" });
+
+            await VerifyCSharpAsync(source, new[] { xmlDictionary, dicDictionary });
+        }
+
+        [Fact]
+        public async Task CorrectWordDisallowedByGlobalXmlDictionary_Verify_EmitsDiagnostic()
+        {
+            var source = "class Program { }";
+            var dictionary = CreateXmlDictionary(null, new[] { "program" });
+
+            await VerifyCSharpAsync(
+                source,
+                dictionary,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.TypeRule)
+                    .WithLocation(1, 7)
+                    .WithArguments("Program", "Program"));
+        }
+
+        [Fact]
+        public async Task MisspellingAllowedByProjectDictionary_Verify_NoDiagnostics()
+        {
+            var source = "class Clazz {}";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(source, dictionary);
+        }
+
+        [Fact(Skip = "Adding additional files to specific projects is not yet supported")]
+        public async Task MisspellingAllowedByDifferentProjectDictionary_Verify_EmitsDiagnostic()
+        {
+            var source = "class Clazz {}";
+            var dictionary = CreateDicDictionary(new[] { "clazz" });
+
+            await VerifyCSharpAsync(
+                source,
+                dictionary,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.TypeRule)
+                    .WithLocation(1, 7)
+                    .WithArguments("Clazz", "Clazz"));
+        }
+
+        [Fact(Skip = "Specifying assembly names is not yet supported")]
+        public async Task AssemblyMisspelled_Verify_EmitsDiagnostic()
+        {
+            var source = "class Program {}";
+
+            await VerifyCSharpAsync(
+                source,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.AssemblyRule)
+                    .WithArguments("Assambly", "MyAssambly"));
+        }
+
+        [Fact]
+        public async Task AssemblyUnmeaningful_Verify_EmitsDiagnostic()
+        {
+            var source = "class Program {}";
+
+            await VerifyCSharpAsync(
+                source,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.AssemblyMoreMeaningfulNameRule)
+                    .WithArguments("A"));
+        }
+
+        [Fact]
+        public async Task NamespaceMisspelled_Verify_EmitsDiagnostic()
+        {
+            var source = "namespace Tests.MyNarmspace {}";
+
+            await VerifyCSharpAsync(
+                source,
+                VerifyCS.Diagnostic(IdentifiersShouldBeSpelledCorrectlyAnalyzer.NamespaceRule)
+                    .WithLocation(1, 17)
```

Comment:
Please don't use this overload, either use the markup notation or `WithSpan`.

Reply:
Switched to markupKey; done